### ISSUE TITLE
Multiselect

### ISSUE THREAD

**Issue by jcabrerazuniga** [AllenFang/react-bootstrap-table#1274]
To enter data in a cell there are several options as checkbox, text, select.... is there a "multi-select" one? The select one allows me to choose just one option from several ones. I need to be able to select some of those options.
Thx

**Comment by AllenFang** [comment]
I've no idea about what you said, you mean in cell editing?

**Comment by jcabrerazuniga** [comment]
yes... At this page:  http://allenfang.github.io/react-bootstrap-table/example.html#celledit
at the Custom Editing Cell Class  part there are tree options: textarea, select, checkbox. Do you have, instead of just  select  a multiselect option?
Thx

**Comment by jcabrerazuniga** [comment]
With the use of React Select (https://jedwatson.github.io/react-select/) I am trying to incorporate a multiselect option to enter data in the table. The code I am using is:

----------------------------------------------------------------------------------------------------
class MultiSelectEd extends React.Component {
  constructor(props) {
    super(props);
    this.updateData = this.updateData.bind(this);
	
    this.state = { selValues: props.defaultValue.selValues };
	
	this.state = { allValues: props.defaultValue.allValues};
	
	console.log('@@@@@@@@@@@@@@@@  prop allValues = ' + this.props.allValues);
	console.log('@@@@@@@@@@@@@@@@ state allValues = ' + this.state.allValues);
  }
  focus() {
    this.refs.inputRef.focus();
  }
  updateData() {
    this.props.onUpdate({ selValues: this.state.selValues });
  }
  render() {
    return (
      <span ref='inputRef' id='id_select_control'>
	    <Select name = "multiSelect-name-intable" options = {this.props.allValues}  value = {this.state.selValues} multi={true} onChange={(ev) => { console.log('+++++++++++++' + ev);}}/>	
			{ /* <select
          value={ this.state.selValues }
          onKeyDown={ this.props.onKeyDown }
          onChange={ (ev) => { this.setState({selValues: ev.currentTarget.value }); } } >
          {  }
        </select> */
        /*<button
          className='btn btn-info btn-xs textarea-save-btn'
          onClick={ this.updateData }>
          save
			</button> */}
      </span>
    );
  }
}

const options = {    
	onAddRow: handleOnAddRow
  };
  
const multiSelectEditor = (onUpdate, props) => (<MultiSelectEd onUpdate={onUpdate } {...props}/>);

-----------------------------------------------------------------------------------------

With this code I've been able to show the control:
![pic1](https://cloud.githubusercontent.com/assets/27739794/25642264/f8dde7ba-2f55-11e7-91ee-e6775654cadf.jpg)

but whenever I try to see the content of the list, its options appear behind the table's rows as in:

![pic2](https://cloud.githubusercontent.com/assets/27739794/25642417/d7dd856a-2f56-11e7-896d-b607392fa975.png)

I am wonder if there is a bug here with the table. I tried z-index but no effects. On the other hand, can you please tell how can I place the selected values in the table at the respective col,row? Your examples are somehow hard to follow to understand the process.

**Comment by AllenFang** [comment]
> at the Custom Editing Cell Class part there are tree options: textarea, select, checkbox. Do you have, instead of just select a multiselect option?

Currently not yet, PR is welcome



> I am wonder if there is a bug here with the table. I tried z-index but no effects

I think so and it was report at before https://github.com/AllenFang/react-bootstrap-table/issues/530

> Your examples are somehow hard to follow to understand the process.

which part?

**Comment by AllenFang** [comment]
let me know if you still have any questions

**Comment by darryl1975** [comment]
Hi Allen,

I m using react-bootstrap-table as I m writing my app in React-TypeScript.

I tried to implement checkbox selecton/multiselection.

Need your help and advice on the following error encountered:

/Users/Darryl/Desktop/project/react/costa/src/components/screens/AnnouncementList.tsx
TypeScript error in /Users/Darryl/Desktop/project/react/costa/src/components/screens/AnnouncementList.tsx(357,102):
No overload matches this call.
  **Overload 1 of 2, '(props: Readonly<BootstrapTableProps>): BootstrapTable', gave the following error.
    Type '{ mode: string; clickToSelect: boolean; onSelect: (row: { [x: string]: string; }, isSelected: any, e: any) => void; onSelectAll: (isSelected: any, rows: string | any[]) => void; }' is not assignable to type 'SelectRow<any>'.
      Types of property 'mode' are incompatible.
        Type 'string' is not assignable to type 'SelectRowMode'.
  Overload 2 of 2, '(props: BootstrapTableProps, context?: any): BootstrapTable', gave the following error.
    Type '{ mode: string; clickToSelect: boolean; onSelect: (row: { [x: string]: string; }, isSelected: any, e: any) => void; onSelectAll: (isSelected: any, rows: string | any[]) => void; }' is not assignable to type 'SelectRow<any>'.  TS2769**

    355 |                                         </style>
    356 |                                         <BootstrapTable data={dataNotBroadcast} striped hover pagination={true} options={options} search={true} multiColumnSearch={true} exportCSV={true} version='4'
  > 357 |                                             headerStyle={{ background: '#9898AD' }} bordered={false} selectRow={ selectRowProp } >
        |                                                                                                      ^
    358 |                                             <TableHeaderColumn dataField="button" dataFormat={cellSelector} width="80" columnClassName="columnClassNameFormat1"></TableHeaderColumn>
    359 |                                             <TableHeaderColumn dataField='id' isKey={true} dataSort={true} width="100" columnClassName="columnClassNameFormat1">S/No.</TableHeaderColumn>
    360 |                                             <TableHeaderColumn dataField='announcementText' dataSort={true} width="900" columnClassName="columnClassNameFormat1">Announcements</TableHeaderColumn>


Thank you so much!
